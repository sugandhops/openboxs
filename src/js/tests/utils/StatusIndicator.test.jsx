import React from 'react';

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import StatusIndicator from 'utils/StatusIndicator';

import '@testing-library/jest-dom';

let statusIndicator;

describe('status indicator', () => {
  beforeEach(() => {
    statusIndicator = (<StatusIndicator variant="test" status="test_status_to_display" />);
  });

  it('should match snapshot', () => {
    const renderedStatusIndicator = renderer.create(statusIndicator);
    expect(renderedStatusIndicator.toJSON()).toMatchSnapshot();
  });

  it('should render properly', () => {
    render(statusIndicator);
    expect(screen.getByTestId('status-indicator')).toBeInTheDocument();
  });

  it('should map status', () => {
    render(statusIndicator);
    expect(screen.getByText('test status to display')).toBeInTheDocument();
  });
});
