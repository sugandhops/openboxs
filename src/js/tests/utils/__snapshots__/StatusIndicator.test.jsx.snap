// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status indicator should match snapshot 1`] = `
<div
  className="d-flex align-items-center justify-content-between"
  data-testid="status-indicator"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <span
      className="test-circle status-circle"
    />
    <span
      className="px-1"
    >
      test status to display
    </span>
  </div>
</div>
`;
